src/order/upper_lower.lean

```diff
@@ -879,6 +926,22 @@ begin
   exact (upper_set.upper _).ord_connected.inter (lower_set.lower _).ord_connected,
 end
 
+lemma is_upper_set_iff_eq_upper_closure :
+  is_upper_set s ↔ s = upper_closure s :=
+⟨λ h, h.upper_closure.symm, λ h, h.symm ▸ upper_set.upper _⟩
+
+lemma is_upper_set_iff_eq_Union_Ici :
+  is_upper_set s ↔ s = ⋃ x ∈ s, Ici x :=
+by { rw [is_upper_set_iff_eq_upper_closure, ← upper_set.infi_Ici, upper_set.coe_infi₂], refl }
```

Comment:
I would drop that one given that this is breaking the interface with `upper_closure`.

Reply:
This lemma is used in `stable_under_specialization_iff_eq_Union`. Is there a good alternative to this lemma to be used in that proof?

Reply:
You could restate `coe_upper_closure` to have `⋃ x ∈ s, Ici x` as the RHS, then `rw [upper_closure_eq, coe_upper_closure]` is your proof.